ddp_experiments.py

```diff
@@ -0,0 +1,282 @@
+import argparse
```

Comment:
Instead of creating many experiment scripts in the root directory, can we wrap them into userbenchmarks and place them in the `userbenchmark/` directory? For example, can we place them in the `userbenchmark/ddp_experiment`, and then users can run the benchmark using `python run_benchmark.py ddp_experiment`?

Reply:
umm, i can move the script under that folder.  but since the script is really just a launcher for submitting many slurm jobs i'm not sure if it's good to try to further wrap it underneath another 'run_benchmark' layer.

Reply:
Wrapping it under `run_benchmark` layer gives many benefits. It automatically gives us ability to run it through Github Actions (such as https://github.com/pytorch/benchmark/actions/workflows/userbenchmark-t4-metal.yml), so that users can use our CI infra to get benchmark results, and it will also connect to internal perf dashboard to track the metrics values over time.

And it keeps the root directory cleaner and more organized as well.

Reply:
@xuzhao9 I moved the scripts from `userbenchmark/distributed` to `userbenchmark/ddp_experiments`. However I have not integrated the experiments with userbenchmark layer. Currently the ddp experiment benchmarks aren't runnable with pytorch as-is, due to some dynamo / pytorch bugs (we need to patch some changes in pytorch to workaround the issue). So we wouldn't be able to benefit from the CI or internal perf dashboard anyway.

Would it be okay to leave the scripts as they are (standalone runnable scripts) for ease of development, until we've fixed those issues in pytorch & dynamo?

Reply:
Yeah, it is okay to me if you prefer not to use userbenchmark/CI feature and only from command line right now.